src/backends/KiboCommerce.js

```diff
@@ -0,0 +1,206 @@
+    });
+    // perform authentication
+    const authTicket = await (
+      await fetch(`https://${this._authHost}/api/platform/applications/authtickets/oauth`, options)
+    ).json();
+    // set expiration time in ms on auth ticket
+    this._calculateTicketExpiration(authTicket);
+    // set authentication ticket on next server runtime object
+    await this._setAuthTicket(authTicket);
+
+    return authTicket;
+  }
+
+  async refreshTicket(kiboAuthTicket) {
+    // create oauth refresh fetch options
+    const options = this._buildFetchOptions({
+      client_id: this._clientId,
+      client_secret: this._sharedSecret,
+      grant_type: 'client_credentials',
+      refresh_token: kiboAuthTicket && kiboAuthTicket.refresh_token,
+    });
+    const refreshedTicket = await (
+      await fetch(`https://${this._authHost}/api/platform/applications/authtickets/oauth`, options)
+    ).json();
+
+    this._calculateTicketExpiration(refreshedTicket);
+
+    await this._setAuthTicket(refreshedTicket);
+
+    return refreshedTicket;
+  }
+
+  async getAccessToken() {
+    // get current Kibo API auth ticket
+
+    let authTicket = await this._getAuthTicket();
+    // if no current ticket, perform auth
+    // or if ticket expired, refresh auth
+    if (!authTicket) {
+      authTicket = await this.authenticate();
+    } else if (authTicket.expires_at < Date.now()) {
+      authTicket = await this.refreshTicket(authTicket);
+    }
+
+    return authTicket.access_token;
+  }
+}
+
+export class KiboCommerce {
+  constructor({ apiHost, authHost, clientId, sharedSecret }) {
+    this.graphQLUrl = `https://${apiHost}/graphql`;
+    this.authHelper = new APIAuthenticationHelper(clientId, sharedSecret, authHost);
+  }
+
+  async getAccessToken() {
+    return await this.authHelper.getAccessToken();
+  }
+
+  async getHeaders() {
+    const authToken = await this.getAccessToken();
+    return {
+      headers: {
+        Authorization: `Bearer ${authToken}`,
+        'Content-Type': 'application/json',
+      },
+    };
+  }
+
+  async performSearch({ query, filter, pageSize, startIndex }) {
+    const headers = await this.getHeaders();
+    const body = {
+      query: productSearchQuery,
+      variables: { query, filter, pageSize, startIndex },
+    };
+    const params = { method: 'POST', ...headers, body: JSON.stringify(body) };
+    const request = await fetch(this.graphQLUrl, params);
+    const response = await request.json();
+    return response.data.productSearch;
+  }
+
+  async getItems(state, ids = []) {
+    const { PAGE_SIZE } = state;
+    const filter = ids.map((id) => `productCode eq ${id}`).join(' or ');
+    try {
+      const result = await this.performSearch({ filter, pageSize: PAGE_SIZE });
+      return this.normalizeItems(result.items);
+    } catch (error) {
+      console.error(error);
+      // throw new ProductSelectorError('Could not get items', ProductSelectorError.codes.GET_SELECTED_ITEMS);
```

Comment:
Why comment this?

Reply:
@rs-amp that was unintentional, sorry about that